presidio-anonymizer/presidio_anonymizer/anonymizers/mask.py

```diff
@@ -1,19 +1,59 @@
 """Mask some or all given text entity PII with given character."""
 from presidio_anonymizer.anonymizers import Anonymizer
+from presidio_anonymizer.anonymizers.validators import validate_parameter
+from presidio_anonymizer.entities import InvalidParamException
 
 
-# TODO implement + test
 class Mask(Anonymizer):
     """Mask the given text with given value."""
 
-    def anonymize(self, original_text=None, params={}):
+    def anonymize(self, text: str = None, params: dict = None) -> str:
         """
-        Anonymize a given amount of text with a given character.
+        Mask a given amount of text with a given character.
 
-        :return: The given text masked as requested
+        :param text: the text to be masked
+        :param params:
+            masking_char: The character to be masked with
+            chars_to_mask: The amount of characters to mask
+            from_end: Whether to mask the text from it's end
+        :return: the masked text
         """
-        # chars_to_replace = params.get("chars_to_replace")
-        # from_end = params.get("from_end")
-        # replace_with = params.get("replace_with")
-        # old_text = params.get("old_text")
-        return ""
+        effective_chars_to_mask = self._get_effective_chars_to_mask(
+            text, params.get("chars_to_mask")
+        )
+        from_end = params.get("from_end")
+        masking_char = params.get("masking_char")
+        return self._get_anonymized_text(
+            text, effective_chars_to_mask, from_end, masking_char
+        )
+
+    def validate(self, params: dict = None) -> None:
+        """
+        Validate the parameters for mask.
+
+        :param params:
+            masking_char: The character to be masked with
+            chars_to_mask: The amount of characters to mask
+            from_end: Whether to mask the text from it's end
+        """
+        masking_char = params.get("masking_char")
+        validate_parameter(masking_char, "masking_char", str)
+        if len(masking_char) > 1:
+            raise InvalidParamException(
+                f"Invalid input, masking_char must be a character"
+            )
+
+        validate_parameter(params.get("chars_to_mask"), "chars_to_mask", int)
+        validate_parameter(params.get("from_end"), "from_end", bool)
+
+    @staticmethod
+    def _get_effective_chars_to_mask(text, chars_to_mask):
+        return min(len(text), chars_to_mask) if chars_to_mask > 0 else 0
```

Comment:
If the user gave 20 chars to mask are we sure this is the desired behavior?
maybe he wants for the test:
04343434343 
20 chars to be returned in the text so no one will think it's a phone number...

Reply:
Aligned with current anonymizer behaviour

Reply:
I think we need to consider this...

Reply:
@omri374 your thought?

Reply:
Omri is on his day off and we need to keep this moving so for now lets keep behavior as is.

Reply:
@SharonHart please open a user story/bug to track this.

Reply:
I don't have any strong opinion here. Shiran has a good point but it can be an improvement we add in the future.

Reply:
@omri374 Consider the option to use mask as default. In that case we don't know the length for the anonymized text so this can help us with passing a large number that will serve every word length.